presidio_evaluator/data_generator/presidio_data_generator.py

```diff
@@ -295,48 +292,119 @@ def name_gendered(row):
+                                         postcode="STREET_ADDRESS",
+                                         secondary_address="STREET_ADDRESS",
+                                         job="TITLE",
+                                         zipcode="ZIP_CODE",
+                                         state_abbr="GPE",
+                                         age="AGE")
+
+    def __init__(self,
+                 locale: str,
+                 lower_case_ratio: float,
+                 additional_entity_providers: List[BaseProvider] = None,
+                 additional_sentence_templates: List[str] = None,
+                 random_seed: SeedType = None):
+        presidio_templates_file_path = raw_data_dir / "templates.txt"
+        self._sentence_templates = PresidioDataGenerator.read_template_file(presidio_templates_file_path)
+        if additional_sentence_templates:
+            self._sentence_templates.extend(additional_sentence_templates)
+
+        presidio_additional_entity_providers = [IpAddressProvider,
+                                                NationalityProvider,
+                                                OrganizationProvider,
+                                                UsDriverLicenseProvider,
+                                                AgeProvider,
+                                                AddressProviderNew,
+                                                PhoneNumberProviderNew,
+                                                ReligionProvider]
+        if additional_entity_providers is None:
+            additional_entity_providers = []
+        additional_entity_providers.extend(presidio_additional_entity_providers)
+
+        fake_person_data_path = raw_data_dir / "FakeNameGenerator.com_3000.csv"
+        fake_person_df = pd.read_csv(fake_person_data_path)
+        fake_person_df = PresidioDataGenerator.update_fake_name_generator_df(fake_person_df)
+        faker = RecordsFaker(records=fake_person_df, locale=locale)
+
+        for entity_provider in additional_entity_providers:
+            faker.add_provider(entity_provider)
+
+        self._data_generator = PresidioDataGenerator(custom_faker=faker, lower_case_ratio=lower_case_ratio)
+        self._data_generator.seed(random_seed)
+        for provider, alias in self.PROVIDER_ALIASES.items():
+            self._data_generator.add_provider_alias(provider_name=provider, new_name=alias)
+
+        self.fake_records = None
+
+    def generate_new_fake_records(self, num_samples: int) -> List[FakerSpansResult]:
+        self.fake_records = list(self._data_generator.generate_fake_data(templates=self._sentence_templates,
+                                                                         n_samples=num_samples))
+        # Map faker generated entity types to Presidio entity types
+        for sample in self.fake_records:
+            for span in sample.spans:
+                span.type = self.faker_to_presidio_entity_type[span.type]
```

Comment:
Perhaps we can add to the faker_to_presidio mapping an additional default value 'other' for cases when a new unmapped key appears, then the mapping will become. What do you think?
```suggestion
                span.type = self.faker_to_presidio_entity_type.get(span.type,'other')
```

Reply:
It makes sense to me! Though would also need to change a bit more than this one line, as we need to track these unknown types and replace them in the template too
I would probably set this as another feature introduced in a new PR with additional tests, even though it's a relatively small change

Reply:
makes sense

Reply:
Opened a new issue to track this https://github.com/microsoft/presidio-research/issues/68